Controllers/CTemperatureController.cpp

```diff
@@ -0,0 +1,299 @@
+/**
+ * @file CTemperatureController.cpp
+ *
+ */
+
+/*
+ * CTemperatureController.cpp
+ *
+ *  Created on: 26 Apr 2022
+ *      Author: salavat.magazov
+ */
+
+#include "CTemperatureController.h"
+#include <stdio.h>
+#include "ICommand.h"
+#include "IHardwareMap.h"
+
+extern IHardwareMap *gp_hardware_map;
```

Comment:
Why not pass this into the class as an argument rather than relying on the linker to figure this out during compile time?

Reply:
Done